### PR TITLE
skip bricks with no path during remove device

### PR DESCRIPTION
**Pull request by raghavendra-talur**

### CODE REVIEW

**Comment by raghavendra-talur** [comment]
WIP, reworking the test.

**Comment by raghavendra-talur** [comment]
@obnoxxx @phlogistonjohn ready for review.

File: apps/glusterfs/device_entry.go [hunk 1]
```diff
@@ -423,14 +425,24 @@ func (d *DeviceEntry) Remove(db wdb.DB,
...
 			volumeEntry, err = NewVolumeEntryFromId(tx, brickEntry.Info.VolumeId)
 			if err != nil {
 				return err
 			}
 			return nil
 		})
 		if err != nil {
-			return err
+			if err == ErrBrickWithEmptyPath {
+				logger.LogError("Skipping brick with empth path, brickID: %v, volumeID: %v, error: %v", brickEntry.Info.Id, brickEntry.Info.VolumeId, err)
```
Comment (phlogistonjohn) [inline apps/glusterfs/device_entry.go]: Perhaps this should be a Warning rather than LogError?

File: apps/glusterfs/device_entry.go [hunk 2]
```diff
@@ -398,6 +398,8 @@ func (d *DeviceEntry) Remove(db wdb.DB,
 	executor executors.Executor,
 	allocator Allocator) (e error) {
 
+	var ErrBrickWithEmptyPath error = fmt.Errorf("Brick has no path")
```
Comment (phlogistonjohn) [inline apps/glusterfs/device_entry.go]: This should probably be a file level constant. If its not meant to be used outside of the scope of this file just lowercase it.

File: apps/glusterfs/device_entry.go [hunk 3]
```diff
@@ -423,14 +425,24 @@ func (d *DeviceEntry) Remove(db wdb.DB,
...
 			return nil
 		})
 		if err != nil {
-			return err
+			if err == ErrBrickWithEmptyPath {
+				logger.LogError("Skipping brick with empth path, brickID: %v, volumeID: %v, error: %v", brickEntry.Info.Id, brickEntry.Info.VolumeId, err)
+				continue
+			} else {
+				return err
+			}
```
Comment (phlogistonjohn) [inline apps/glusterfs/device_entry.go]: (Minor) IIRC its considered more idomatic Go to leave off the else block.

**Pull request merged**